store/src/java/com/zimbra/cs/fb/ExchangeEWSFreeBusyProvider.java

```diff
@@ -920,63 +921,55 @@ public boolean handleMailboxChange(String accountId) {
         }
 
         for (Request re : req) {
-            String fb = "";
+            
             int i = 0;
             for (FreeBusyResponseType attendeeAvailability : results) {
-            	if (attendeeAvailability.getFreeBusyView() != null) {
-                	String fbResponseViewType = attendeeAvailability.getFreeBusyView().getFreeBusyViewType().get(0);
-                	String emailAddress = attendees.getMailboxData().get(i).getEmail().getAddress();
-                	String name = attendees.getMailboxData().get(i).getEmail().getName();
-                	ZimbraLog.fb.debug("Exchange EWS free busy view type from response : %s", fbResponseViewType);
+                if (attendeeAvailability.getFreeBusyView() != null) {
+                    	String fbResponseViewType = attendeeAvailability.getFreeBusyView().getFreeBusyViewType().get(0);
+                    	String emailAddress = attendees.getMailboxData().get(i).getEmail().getAddress();
+                    	String name = attendees.getMailboxData().get(i).getEmail().getName();
+                    	ZimbraLog.fb.debug("For user :%s free busy response type received is : %s", emailAddress, fbResponseViewType);
+                	
                     if (re.email == emailAddress) {
-                        if (ResponseClassType.SUCCESS != attendeeAvailability.getResponseMessage().getResponseClass()) {
-                            ZimbraLog.fb.warn("%s :: %s",attendeeAvailability.getResponseMessage().getResponseCode(), attendeeAvailability.getResponseMessage().getMessageText());
+                        if (ResponseClassType.ERROR == attendeeAvailability.getResponseMessage().getResponseClass()) {
+                            ZimbraLog.fb.debug("Unable to fetch free busy for %s  error code %s :: %s",
+                                emailAddress,
+                                attendeeAvailability.getResponseMessage().getResponseCode(),
+                                attendeeAvailability.getResponseMessage().getMessageText());
                             if (attendeeAvailability.getResponseMessage().getResponseCode().equals(ResponseCodeType.ERROR_NO_FREE_BUSY_ACCESS)) {
-                            	FreeBusy npFreeBusy = FreeBusy.nodataFreeBusy(name, req.get(0).start, req.get(0).end);
-                            	npFreeBusy.setHasPermission(false);
-                            	ret.add(npFreeBusy);
+                                	FreeBusy npFreeBusy = FreeBusy.nodataFreeBusy(name, req.get(0).start, req.get(0).end);
```

Comment:
req.get(0).start and  req.get(0).end has been used at several places. better to assign these values on top and use the variables.